libbeat/processors/move_fields/move_fields_test.go

```diff
@@ -0,0 +1,105 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package move_fields
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+func TestMoveFields(t *testing.T) {
+	cases := []struct {
+		in, expected mapstr.M
+		p            *moveFields
+	}{
+		{
+			mapstr.M{"app": mapstr.M{"version": 1, "method": "2"}, "other": 3},
+			mapstr.M{"app": mapstr.M{"method": "2"}, "rpc": mapstr.M{"version": 1}, "other": 3},
+			&moveFields{config: moveFieldsConfig{
+				ParentPath: "app",
+				From:       nil,
+				To:         "rpc.",
+				Exclude:    []string{"method"},
+				excludeMap: map[string]struct{}{"method": {}},
+			}},
+		},
```

Comment:
Using the input and expected result as name for the tests here does now work very well, so it's better to add a `name` field to your test case and use it when running the test. The name can be a short description of what you're testing on that case:
```suggestion
	cases := []struct {
		name         string
		in, expected mapstr.M
		p            *moveFields
	}{
		{
			"move fields to a new object",
			mapstr.M{"app": mapstr.M{"version": 1, "method": "2"}, "other": 3},
			mapstr.M{"app": mapstr.M{"method": "2"}, "rpc": mapstr.M{"version": 1}, "other": 3},
			&moveFields{config: moveFieldsConfig{
				ParentPath: "app",
				From:       nil,
				To:         "rpc.",
				Exclude:    []string{"method"},
				excludeMap: map[string]struct{}{"method": {}},
			}},
		},
```